dialer_test.go

```diff
@@ -0,0 +1,155 @@
+package cloudsqlconn
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"cloud.google.com/cloudsqlconn/internal/mock"
+	"google.golang.org/api/option"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+// replaceSQLAdminClient patches the Dialer's SQL Admin API client with one that
+// uses the provided HTTP client and endpoint.
+func replaceSQLAdminService(d *Dialer, client *http.Client, endpoint string) {
```

Comment:
Is this just to save some boilerplate for converting `mc, url` into a service? If so, we should consider having a function that just skips a step and returns the SQLAdminService